src/Customer/Service.php

```diff
@@ -31,29 +31,29 @@ public function find(string $uid): Customer;
     /**
      * Associates a card to a customer.
      *
-     * @param Customer $customer
-     * @param Card $card
+     * @param string $customerUid
+     * @param string $cardUid
      *
      * @return Card
      */
-    public function addCard(Customer $customer, Card $card): Card;
+    public function addCard(string $customerUid, string $cardUid): Card;
 
     /**
      * Deletes a customers card.
      *
-     * @param Customer $customer
-     * @param Card $card
+     * @param string $customerUid
+     * @param string $cardUid
      *
      * @return void
      */
-    public function deleteCard(Customer $customer, Card $card): void;
+    public function deleteCard(string $customerUid, string $cardUid): void;
 
     /**
      * Retrieves a customers cards.
      *
-     * @param Customer $customer
+     * @param string $uid
```

Comment:
Event though I don't approve the removal of objects over primitive types, we don't know what kind of identifier this is?

Would it be possible to have it closer to a Customer class rather than a Customer Service class? This way you could call it like:
```php
$foobar->customers()->find($customerUid)->addCard($cardUid);
```
directly on a Customer object as it's an action performed for a Customer.

Reply:
This is the Customer Service so I don't think we need to specify what type of $uid is needed. We are specifying the type of uid needed if there are multiple uid in used in the same method.

Reply:
To add to what @adteulade said @clemblanco we opted for all the "doing" (creating/retrieving/updating/deleting) to be within the services so that the models can remain as plain old php objects.

But also I wouldn't be against this named `customerUid` just to be explicit.

Reply:
> we opted for all the "doing" (creating/retrieving/updating/deleting) to be within the services
Could still happen with fluent methods.

Reply:
@clemblanco First sorry for being annoying with that but:
Based on:

> ... so that the models can remain as plain old php objects.

If we are using:

> $foobar->customers()->find($customerUid)->addCard($cardUid);

When calling `find()` we'll get a Customer object which mean having `addCards()` using the Customer model will imply that models contains logic code instead of being plain old php objects.